src/main/java/at/hannibal2/skyhanni/features/event/spook/TheGreatSpook.kt

```diff
@@ -53,11 +60,70 @@ object TheGreatSpook {
     }
 
+    /**
+     * REGEX-TEST: §d§lQUICK MATHS! §r§7Solve: §r§e(10*2)+12*5
+     */
+    private val mathFearMessagePattern by RepoPattern.pattern(
+        "chat.math",
+        "§d§lQUICK MATHS! §r§7Solve: §r§e(?<math>.*)",
+    )
+
+    /**
+     * REGEX-TEST: §4[FEAR] Public Speaking Demon§r§f: Speak PlasticEating!
+     */
+    private val speakingFearMessagePattern by RepoPattern.pattern(
+        "chat.speaking",
+        "§4\\[FEAR] Public Speaking Demon§r§f: (Speak|Say something interesting) (?<name>.*)!",
+    )
+
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (isMathSolverEnabled()) {
+            mathFearMessagePattern.matchMatcher(event.message) {
+                val math = group("math")
+                val result = NEUCalculator.calculateOrNull(math)?.toInt()
+                DelayedRun.runNextTick {
+                    if (result != null) {
+                        ChatUtils.clickToActionOrDisable(
+                            "The result is: $result",
+                            option = config.primalFearSolver::math,
+                            actionName = "Send the result",
+                            action =
+                            {
+                                HypixelCommands.allChat(result.toString())
+                            },
+                        )
+                    } else {
+                        ChatUtils.chat("An unexpected error occurred while solving the math problem.")
+                    }
+                }
+            }
+        }
+
+        if (isSpeakingSolverEnabled()) {
+            speakingFearMessagePattern.matchMatcher(event.message) {
+                DelayedRun.runNextTick {
+                    val feature: KMutableProperty0<*> = config.primalFearSolver::publicSpeaking
+                    ChatUtils.clickToActionOrDisable(
+                        "Click to send a random string to complete the Primal Fear",
+                        feature,
+                        "send a random string.",
+                        action =
+                        {
+                            HypixelCommands.allChat("I looove SkyHanni! ${StringUtils.generateRandomString(4)}")
+                        }
+                    )
+                }
+            }
+        }
+    }
+
     private fun isTimerEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearTimer
 
     private fun isNotificationEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearNotification
     private fun isFearStatEnabled(): Boolean = LorenzUtils.inSkyBlock && config.fearStatDisplay
     private fun isTimeLeftEnabled(): Boolean = LorenzUtils.inSkyBlock && config.greatSpookTimeLeft
 
-    private fun isAllDisabled(): Boolean = !isTimeLeftEnabled() && !isTimerEnabled() && !isFearStatEnabled() && !isNotificationEnabled()
+    private fun isMathSolverEnabled(): Boolean = LorenzUtils.inSkyBlock && config.primalFearSolver.math
```

Comment:
remove the two functions: check if in sb only once in the event and inline the two feature checks

Reply:
only my 2? or every function